framework/CodeInterfaces/PHISICS/DecayParser.py

```diff
@@ -0,0 +1,172 @@
+          line[0] = "{0:<7s}".format(line[0])
+          i = 1
+          while i <= len(self.decayModeNumbering[atomicNumber]):
+            line[i] = "{0:<11s}".format(line[i])
+            i = i + 1
+          outfile.writelines(' '+''.join(line[0:len(self.decayModeNumbering[atomicNumber]) + 1])+"\n")
+      except KeyError: # happens for all the unperturbed isotopes
+        pass
+
+  def hardcopyPrinter(self,atomicNumber,modifiedFile):
+    """
+      Prints the hardcopied information at the begining of the xml file.
+      @ In, atomicNumber, integer, indicates if the isotope parsed is an actinide (0) or a fission product (1)
+      @ In, modifiedFile, string, output temperary file name
+      @ Out, None
+    """
+    flag = 0
+    with open(modifiedFile, 'a') as outfile:
+      with open(self.inputFiles) as infile:
+        for line in infile:
+          if not line.split():
+            continue   # if the line is blank, ignore it
+          if re.match(r'(.*?)'+atomicNumber+'s',line.strip()) and atomicNumber == self.isotopeParsed[0]:
+            flag = 2
+          if flag == 2:
+            if re.match(r'(.*?)\s+\w+(\W)\s+\w+(\W)',line) and any(s in 'BETA' for s in line.split()) and atomicNumber == self.isotopeParsed[0]:
+              outfile.writelines(line)
+              break
+            outfile.writelines(line)
+          if any(s in atomicNumber+'roducts' for s in line.split()):
+            flag = 1
+          if  flag == 1:
+            if re.match(r'(.*?)\s+\w+(\W)\s+\w+(\W)',line) and any(s in 'BETA' for s in line.split()) and atomicNumber == self.isotopeParsed[1]:
+              outfile.writelines(line)
+              break
+            outfile.writelines(line)
+        self.matrixPrinter(infile, outfile, atomicNumber)
+
+  def __init__(self,inputFiles,workingDir,**pertDict):
+    """
+      @ In, inputFiles, string, xml Decay file.
+      @ In, workingDir, string, absolute path to the working directory
+      @ In, pertDict, dictionary, dictionary of perturbed variables
+      @ Out, None
+    """ 
+    self.pertDict = pertDict
+    self.inputFiles = inputFiles
+    for key, value in self.pertDict.iteritems():
+      self.pertDict[key] = '%.3E' % Decimal(str(value)) #convert the values into scientific values
+    numbering = {}
+    concatenateDecayList = []
+    self.allDecayList = []
+    OpenInputFile = open (self.inputFiles, "r")
+    lines = OpenInputFile.readlines()
+    OpenInputFile.close()
+    self.inputFiles = inputFiles
+    self.isotopeClassifier = {}   # FP or Actinide
+    self.isotopeParsed=['Actinide','FP']
+    self.decayModeNumbering = {}
+    for line in lines:
+      if   re.match(r'(.*?)Actinides', line):
+        typeOfIsotopeParsed = self.isotopeParsed[0]
+      elif re.match(r'(.*?)FProducts', line):
+        typeOfIsotopeParsed = self.isotopeParsed[1]
+      if (re.match(r'(.*?)\w+(\W?)\s+\w+(\W?)\s+\w',line) and any(s in "BETA" for s in line)) : # create dynamic column detector
```

Comment:
the search for 'BETA' is an additional strength to make sure the line looked up is the 'label line':
_BETA BETA+ ALPHA [...]_
If I look only for  \w+(\W?)\s+\w+(\W?)\s+\w, any two consecutif words would be a match. 
If I look for N consecutive words, N>2 (which is the case in the label line right now), the case where columns are deleted and less than N, it would error out. 
Right now, the codes errors out (or bug) if there are less than two decay columns, or if the decay type does not exist. 
That's the most solid option I saw.